spiketoolkit/sorters/tridesclous/tridesclous.py

```diff
@@ -1,95 +1,125 @@
-                        use_sparse_template=use_sparse_template,
-                        sparse_threshold_mad=1.5,
-                        use_opencl_with_sparse=use_opencl_with_sparse,)
-    peeler.run(duration=None, progressbar=True)
+    def _setup_recording(self):
+        # reset the output folder
+        if os.path.exists(self.output_folder):
+            shutil.rmtree(self.output_folder)
+        
+        os.makedirs(self.output_folder)
+
+        # save prb file:
+        probe_file = self.output_folder / 'probe.prb'
+        se.saveProbeFile(self.recording, probe_file, format='spyking_circus')
+        
+        # save binary file
+        raw_filename = self.output_folder / 'raw_signals.raw'
+        traces = self.recording.getTraces()
+        dtype = traces.dtype
+        with open(raw_filename, mode='wb') as f:
+            f.write(traces.T.tobytes())
+        
+        # initialize source and probe file
+        self.tdc_dataio = tdc.DataIO(dirname=output_folder)
+        nb_chan = self.recording.getNumChannels()
+        
+        self.tdc_dataio.set_data_source(type='RawData', filenames=[raw_filename],
+                        dtype=dtype.str, sample_rate=self.recording.getSamplingFrequency(),
+                                        total_channel=nb_chan)
+        self.tdc_dataio.set_probe_file(probe_file)
+        if self.debug:
+            print(self.tdc_dataio)        
+        
+    def _run(self):
+        nb_chan = self.recording.getNumChannels()
     
-    sorting = se.TridesclousSortingExtractor(tdc_dirname)
-    return sorting
+        # check params and OpenCL when many channels
+        use_sparse_template = False
+        use_opencl_with_sparse = False
+        if nb_chan >64: # this limit depend on the platform of course
+            if tdc.cltools.HAVE_PYOPENCL:
+                # force opencl
+                self.params['fullchain_kargs']['preprocessor']['signalpreprocessor_engine'] = 'opencl'
+                use_sparse_template = True
+                use_opencl_with_sparse = True
+            else:
+                print('OpenCL is not available processing will be slow, try install it')
+        
+        # make catalogue
+        # TODO check which channel_group
+        cc = tdc.CatalogueConstructor(dataio=self.tdc_dataio, chan_grp=0)
+        tdc.apply_all_catalogue_steps(cc, verbose=debug, **self.params)
+        if self.debug:
+            print(cc)
+        cc.make_catalogue_for_peeler()
+        
+        # apply Peeler (template matching)
+        initial_catalogue = self.tdc_dataio.load_catalogue(chan_grp=0)
+        peeler = tdc.Peeler(self.tdc_dataio)
+        peeler.change_params(catalogue=initial_catalogue,
+                            use_sparse_template=use_sparse_template,
+                            sparse_threshold_mad=1.5,
+                            use_opencl_with_sparse=use_opencl_with_sparse,)
+        peeler.run(duration=None, progressbar=True)
+
+
+def run_tridesclous(
```

Comment:
So this is a different function then run_sorter_engine?

Reply:
Should they be the same?

Reply:
At this moment it is the same.
I was wondering between a unique generic one or one runner function for each SorerClass.

The run_sorter_engine have also in the signautre the SorterClass itself.

the run_tridesclous have the same signeture as beefore (and so for run_klusta)

Reply:
I think each algorithm should have the same function, "run_sorter". The name and signature for this function should be the same across spike sorters. Is that possible? It would make running all sorters the same which is great for standardized analysis.